uamqp/_connection.py

```diff
@@ -204,11 +217,18 @@ def _send_frame(self, channel, frame, timeout=None, **kwargs):
         :rtype: None
         """
         if self._can_write():
-            self._last_frame_sent_time = time.time()
-            if timeout:
-                with self._transport.block_with_timeout(timeout):
-                    self._transport.send_frame(channel, frame, **kwargs)
-            self._transport.send_frame(channel, frame, **kwargs)
+            try:
+                self._last_frame_sent_time = time.time()
+                if timeout:
+                    with self._transport.block_with_timeout(timeout):
+                        self._transport.send_frame(channel, frame, **kwargs)
+                self._transport.send_frame(channel, frame, **kwargs)
+            except Exception as exc:
+                self._error = AMQPConnectionError(
+                    ErrorCodes.InternalError,
```

Comment:
yes, this should be catching socket errors.

The only error I have encountered is "ConnectionResetError : [winerror 10054] an existing connection was forcibly closed by the remote host" which happened due to 4 mins idle timeout.

Apart from that I see that in our transport.py there're errors like `SSLError` or `socket.error` that could be raised.

Are you talking about introduce a new error type? -- to me I'm treating socket error as kind of AMQPConnectionError..

Reply:
I didn't mean a new error type - rather we should not use the code 'InternalError' - it's a bit ambiguous.

Reply:
I agree with you.

I was trying to find if there's any shared amqp error condition code we could use, and "internal error" is a general one that could fit -- and that's what uamqp-c does.

honestly, I'm not sure how much freedom we have here in defining condition code. But since this is our own implementation, and the client side will take care of the error, I think probably it's fine to define an "amqp:socket-error" condition?

apart from the condition code itself, I don't think the error handling logic will change as the connection needs to be reestablished anyway.